nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-standard-record-utils/src/main/java/org/apache/nifi/schema/access/HortonworksAttributeSchemaReferenceWriter.java

```diff
@@ -22,16 +22,29 @@
 
 import java.io.IOException;
 import java.io.OutputStream;
-import java.util.EnumSet;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
 
 public class HortonworksAttributeSchemaReferenceWriter implements SchemaAccessWriter {
-    private static final Set<SchemaField> requiredSchemaFields = EnumSet.of(SchemaField.SCHEMA_IDENTIFIER, SchemaField.SCHEMA_VERSION);
-    static final int LATEST_PROTOCOL_VERSION = 1;
+
     static final String SCHEMA_BRANCH_ATTRIBUTE = "schema.branch";
 
+    private final Integer protocolVersion;
+
+    public HortonworksAttributeSchemaReferenceWriter(final Integer protocolVersion) {
```

Comment:
If `protocolVersion` cannot be null, is there a reason to make it an `Integer` instead of an `int`?